2019-03-Autumn/2_string_manipulation/palindrome_permutation/Palindrome_Test.java

```diff
@@ -0,0 +1,182 @@
+
+public class Palindrome_Test {
+    /**
+     * Solution #1
+     */
+    static boolean isPermutationOfPalindrome(String phrase) {
+        int[] table = buildCharFrequencyTable(phrase);
+        return checkMaxOneOdd(table);
+    }
+
+    /* Check that no more than one character has an odd count. */
+    static boolean checkMaxOneOdd(int[] table) {
+        boolean foundOdd = false;
+        for(int count:table) {
+            if(count % 2 == 1) {
+                if(foundOdd) {
+                    return false;
+                }
+                foundOdd = true;
+            }
+        }
+        return true;
+    }
+
+    /* Count how many times each character appears. */
+     static int[] buildCharFrequencyTable(String phrase) {
+        int[] table = new int[Character.getNumericValue('z') - Character.getNumericValue('a') + 1];
+        for(char c : phrase.toCharArray()) {
+            int x = getCharacterNumber(c);
+            if(x != -1) {
+                table[x]++;
+            }
+        }
+        return table;
+    }
+
+
+    /**
+     * Solution #2
+     */
+//    static boolean isPermutationOfPalindrome(String phrase) {
+//        int countOdd = 0;
+//        int[] table = new int[Character.getNumericValue('z') - Character.getNumericValue('a') + 1];
+//        for(char c : phrase.toCharArray()) {
+//            int x = getCharNumber(c);
+//            if(x != -1) {
+//                table[x]++;
+//                if(table[x] % 2 == 1) {
+//                    countOdd++;
+//                } else {
+//                    countOdd--;
+//                }
+//            }
+//        }
+//        return countOdd <= 1;
+//    }
```

Comment:
Instead of commenting out solution functions, I'd recommend using a different naming convention for methods when you include multiple implementations, eg appending the function name with `_SolutionX`:

```java
static boolean isPermutationOfPalindrome_Solution1(String phrase) {}
static boolean isPermutationOfPalindrome_Solution2(String phrase) {}
```